### PR TITLE
05 view a single review

### PR DESCRIPTION
**Pull request by HappyJuice123**

### CODE REVIEW

**Comment by HappyJuice123** [comment]
the colour are a mess right now, I'll sort out the colours after this PR

File: src/Components/FeaturedReviews.jsx [hunk 1]
```diff
@@ -15,19 +15,32 @@ export const FeaturedReviews = ({ reviews }) => {
         <h2>Featured Reviews</h2>
```
Comment (JVal-NC) [inline src/Components/FeaturedReviews.jsx:9]: Line 4: considering extracting the logic into another file.

File: src/Components/FeaturedReviews.jsx [hunk 2]
```diff
@@ -15,19 +15,32 @@ export const FeaturedReviews = ({ reviews }) => {
...
-                <h3>{reviews[index].title}</h3>
-                <img
-                  src={reviews[index].review_img_url}
-                  alt={reviews[index].title}
-                  className="review-img"
-                />
+                <section>
+                  <Link
+                    to={`/reviews/${reviews[index].review_id}`}
+                    id="review-title"
```
Comment (JVal-NC) [inline src/Components/FeaturedReviews.jsx]: id should be unique, className can be used for CSS if multiple tags require same styling

File: src/Components/Header.jsx [hunk 3]
```diff
@@ -2,9 +2,11 @@ import { Link } from "react-router-dom";
 export const Header = () => {
   return (
     <header>
-      <Link to="/">
-        <h1 className="header">Games Review</h1>
-      </Link>
+      <h1>
+        <Link to="/" id="header">
```
Comment (JVal-NC) [inline src/Components/Header.jsx]: h1 should probably be the child of the <Link>

File: src/App.js [hunk 4]
```diff
@@ -5,12 +5,16 @@ import { Reviews } from "./Components/Reviews";
 import { useEffect, useState } from "react";
 import { getReviews } from "./utils";
 import { FeaturedReviews } from "./Components/FeaturedReviews";
+import { SingleReview } from "./Components/SingleReview";
 
 function App() {
   const [reviews, setReviews] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
```
Comment (JVal-NC) [inline src/App.js]: Consider providing individual laoding state for each component fetching data from the api. Subsequently do your reviews need to be held in state within App.js at this stage

File: src/Components/Reviews.jsx [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
-export const Reviews = ({ reviews }) => {
+export const Reviews = ({ reviews, isLoading }) => {
   return (
     <main>
       <header>
         <h2>Reviews</h2>
       </header>
-      <ReviewCard reviews={reviews} />
+      <section>
+        <ReviewCard reviews={reviews} isLoading={isLoading} />
```
Comment (JVal-NC) [inline src/Components/Reviews.jsx]: Feel ReviewCard could be confusingly named when it represents multiple reviews opposed to a single one

File: src/Components/ReviewCard.jsx [hunk 6]
```diff
@@ -1,21 +1,31 @@
...
-              <h3>{review.title}</h3>
-              <img
-                src={review.review_img_url}
-                alt={review.title}
-                className="review-img"
-              />
+              <section>
+                <Link
+                  to={`/reviews/${review.review_id}`}
+                  id="review-card__link"
```
Comment (JVal-NC) [inline src/Components/ReviewCard.jsx]: Similar issue here to previous comment with the id not being unique to each tag.

File: src/Components/ReviewCard.jsx [hunk 7]
```diff
@@ -1,21 +1,31 @@
...
+                >
+                  <h3 className="review-title">{review.title}</h3>
+                  <img
+                    src={review.review_img_url}
+                    alt={review.title}
+                    className="review-img"
+                  />
+                </Link>{" "}
```
Comment (JVal-NC) [inline src/Components/ReviewCard.jsx]: Is this space intentional? Did prettier put it in?

File: src/Components/SingleReview.jsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+  useEffect(() => {
+    setIsLoading(true);
+    getReviewById(review_id)
+      .then((data) => {
+        setReviewExists(true);
+        setSingleReview(data.review);
+        setIsLoading(false);
+      })
+      .catch(() => {
```
Comment (JVal-NC) [inline src/Components/SingleReview.jsx:20]: loading state will need to be flipped in order to see your UI reflect your error state - however there could be another error other than a nonexistent review. Your backend is giving you the right errors, feel free to come back to this at a later ticket.

File: src/Components/SingleReview.jsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        setReviewExists(false);
+      });
+  }, [review_id]);
+
+  return !isLoading ? (
+    reviewExists ? (
+      <main className="single-review">
+        <section>
+          <Link to="/reviews" id="single-review-to-reviews">
```
Comment (JVal-NC) [inline src/Components/SingleReview.jsx]: Reminder about id prop needing to be unique

File: src/Components/SingleReview.jsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+          <p id="single-review__votes">Votes: {singleReview.votes}</p>
+          <p id="single-review__comment-count">
+            {singleReview.comment_count} Comments
+          </p>
+          {singleReview.created_at ? (
+            <p id="single-review__created">
+              {singleReview.owner} on {singleReview.created_at.substring(0, 10)}
+            </p>
+          ) : (
+            <p>{""}</p>
```
Comment (JVal-NC) [inline src/Components/SingleReview.jsx:61]: If you do want to show something in the event of no created_at property, put a placeholder here for now.

However do all your reviews not have created_at applied?

**Review by JVal-NC — COMMENTED** [review]
Approved PR, but please look at the comments and let us know if you have any questions

**Pull request merged**